pkg/datamover/profile.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2023 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datamover
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	crv1alpha1 "github.com/kanisterio/kanister/pkg/apis/cr/v1alpha1"
+	"github.com/kanisterio/kanister/pkg/kopia/snapshot"
+	"github.com/kanisterio/kanister/pkg/param"
+)
+
+type Profile struct {
+	OutputName string
+	Profile    *param.Profile
+	SnapJSON   string
+}
+
+func (p *Profile) Pull(sourcePath, destinationPath string) error {
+	ctx := context.Background()
```

Comment:
We should add a context parameter to methods in the Datamover interface (Pull, Push, Delete). Propagate context from caller instead of creating new.

Reply:
Done !